src/ui/public/courier/fetch/request/segmented.js

```diff
@@ -1,334 +0,0 @@
-import _ from 'lodash';
-import { Notifier } from 'ui/notify/notifier';
-import { SearchRequestProvider } from './search_request';
-import { SegmentedHandleProvider } from './segmented_handle';
-
-export function SegmentedRequestProvider(Private, timefilter, config) {
-  const SearchRequest = Private(SearchRequestProvider);
-  const SegmentedHandle = Private(SegmentedHandleProvider);
-
-  const notify = new Notifier({
-    location: 'Segmented Fetch'
-  });
-
-  class SegmentedReq extends SearchRequest {
-    constructor(source, defer, initFn) {
-      super(source, defer);
-
-      this.type = 'segmented';
-
-      // segmented request specific state
-      this._initFn = initFn;
-
-      this._desiredSize = null;
-      this._maxSegments = config.get('courier:maxSegmentCount');
```

Comment:
Crap, I forgot about this setting. I was thinking we could make this change in 6.x since it only affects implementation details. But the fact that we're exposing a setting to tune those details may change things. What do you think @epixa ?

Reply:
So, it does still have an effect before this change. It sets a max on the number of requests a search can be segmented into. After this change, searches are no longer broken up into segments, so the setting is useless. I can't think of a situation where segmentation would actually be useful (slow queries maybe, but to me that's a bug in ES or the request itself), but in theory someone might be relying on it for reasons unknown to me at this time.

Reply:
In that case, this does seem like a breaking change to me. Regardless of the use case, at some point we felt there was enough of a reason to give people control over their segmenting behavior. Our lack of memory or knowledge on that shouldn't be enough of a reason to just remove it. It seems like we should target master and remove all of this alongside the interval index pattern support itself.

Reply:
Went ahead and removed this setting since I'm only targeting 7.0 now.